### ISSUE TITLE
Crash when first link in file is ignored/replaced

### ISSUE THREAD

**Issue by nschonni** [tcort/markdown-link-check#50]
I think this is really for the `link-check` project, but i'll post it here since this is the one it bubbled up in.
I believe this is occurring because there are some cases where the `aliveStatusCodes` doesn't get initialized properly before this is called.

```console
/usr/local/lib/node_modules/markdown-link-check/node_modules/link-check/lib/LinkCheckResult.js:8
        this.status = opts.aliveStatusCodes.some((statusCode) => (statusCode instanceof RegExp) ? statusCode.test(this.statusCode) : statusCode === this.statusCode) ? 'alive' : 'dead';
                                            ^

TypeError: Cannot read property 'some' of undefined
    at new LinkCheckResult (/usr/local/lib/node_modules/markdown-link-check/node_modules/link-check/lib/LinkCheckResult.js:8:45)
    at /usr/local/lib/node_modules/markdown-link-check/index.js:33:32
    at /usr/local/lib/node_modules/markdown-link-check/node_modules/async/dist/async.js:1135:9
    at replenish (/usr/local/lib/node_modules/markdown-link-check/node_modules/async/dist/async.js:1011:17)
    at /usr/local/lib/node_modules/markdown-link-check/node_modules/async/dist/async.js:1016:9
    at _asyncMap (/usr/local/lib/node_modules/markdown-link-check/node_modules/async/dist/async.js:1133:5)
    at Object.mapLimit (/usr/local/lib/node_modules/markdown-link-check/node_modules/async/dist/async.js:1219:16)
    at markdownLinkCheck (/usr/local/lib/node_modules/markdown-link-check/index.js:26:11)
    at runMarkdownLinkCheck (/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:89:5)
    at ReadStream.<anonymous> (/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:74:21)
```

**Comment by timmkrause** [comment]
Yeah.

`opts.aliveStatusCodes = opts.aliveStatusCodes || [ 200 ];`

is called during the actual link check. If the first link now is ignored this isn't set and when `LinkCheckResult` is used to return the `ignored` result it evaluates `opts.aliveStatusCodes` which is undefined.

**Comment by tcort** [comment]
Thanks for the report and the fix! I've published v3.6.2 which should solve this issue.
